doc/feature/synapses.md

```diff
@@ -0,0 +1,189 @@
+Synapse support in Brain
+============
+
+The low-level data access for synapses provided by Brion needs a complementary
+API in Brain to support various common higher-level uses cases, like semantic
+synapse object access or full-circuit synapse density voxelization, to name the
+two extremes.
+
+## Requirements
+
+* Semantic synapse objects that give read-access to all properties
+* Array access which allows vectorization
+* Filtering of synapses for pathway use case (shared synapses of two GIDSets)
+* Lazy-loading to support out-of-core scenarios
+* User-transparent KVS caching
+
+## New dependencies
+
+None
+
+## API
+
+```cpp
+
+class Circuit
+{
+    SynapsesStream getAfferentSynapses( const GIDSet& gids ) const;
+    SynapsesStream getEfferentSynapses( const GIDSet& gids ) const;
+    SynapsesStream getSharedSynapses( const GIDSet& preGIDs,
+                                      const GIDSet& postGIDs ) const;
```

Comment:
I guess this functions always load the data and don't cache anything internally, so if I want to do some operations with targets I should keep the containers myself, right?
This also means that if I want to have quick access to the afferent and efferent sets of each cell involved in a projection (like BBPSDK provides), I'd have to call getSharedSynapses pre.size() + post.size() times
(first each pre cell against the full post set for the afferent synapses, and then the reverse for the efferent), which is not a big issues, but then, much more importantly, there's no way the implementation can avoid having each synapse twice in memory.